src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -281,20 +281,38 @@
     (future-cancel command-loader))
   context)
 
-(defn initialize-schema
-  "Ensure the database is migrated to the latest version, and warn if
-  it's deprecated, log and exit if it's unsupported.
-
-  This function will return true iff any migrations were run."
-  [db-conn-pool config]
+(defn db-unsupported-msg
+  "Returns a message describing which databases are supported."
+  []
+  (str (trs "PostgreSQL DB versions older than {0}.{1} are no longer supported."
+            (first scf-store/oldest-supported-db)
+            (second scf-store/oldest-supported-db))
+       " "
+       (trs "Please upgrade Postgres and restart PuppetDB.")))
+
+(defn shutdown-if-db-not-supported [shutdown-on-error]
+  (let [oldest scf-store/oldest-supported-db]
```

Comment:
Sure, though I wondered if it might make sense to track that dividing line independently, and in the storage layer -- but I don't feel strongly about it.